src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -121,7 +123,13 @@
     /**
      * What he currently might be needing.
      */
-    protected Predicate<ItemStack> needsCurrently = null;
+    protected Tuple<Predicate<ItemStack>, Integer> needsCurrently = null;
```

Comment:
Yes the Tuple is needed because only the Predicate is inside. Predicate<ItemStack>. this doesnt contain the count. the Tuple <Predicate<ItemStack>, Integer> was suggested by Raycoms.